compiler/src/dotty/tools/dotc/core/Decorators.scala

```diff
@@ -246,13 +247,30 @@ object Decorators {
       }
 
   extension [T](x: T)
-    def showing(
-        op: WrappedResult[T] ?=> String,
-        printer: config.Printers.Printer = config.Printers.default): T = {
-      printer.println(op(using WrappedResult(x)))
+    def showing[U](
+        op: WrappedResult[U] ?=> String,
+        printer: config.Printers.Printer = config.Printers.default)(using c: Conversion[T, U] = null): T = {
+      // either the use of `$result` was driven by the expected type of `Shown`
+      // which lead to the summoning of `Conversion[T, Shown]` (which we'll invoke)
+      // or no such conversion was found so we'll consume the result as it is instead
+      val obj = if c == null then x.asInstanceOf[U] else c(x)
+      printer.println(op(using WrappedResult(obj)))
       x
     }
 
+    /** Instead of `toString` call `show` on `Showable` values, falling back to `toString` if an exception is raised. */
+    def show(using Context)(using z: Show[T] = null): String = x match
```

Comment:
This method has two modes of operation depending on if the implicit is found or not. This seems a bit dangerous as at call site the semantics (generated string) might change silently. I would propose to be more explicit and have two methods 
```scala
    def show(using Context)(using z: Show[T]): String =
      z.show(x).toString

    def showAny(using Context): String = x match
      case x: Showable =>
        try x.show
        catch
          case ex: CyclicReference => "... (caught cyclic reference) ..."
          case NonFatal(ex)
              if !ctx.mode.is(Mode.PrintShowExceptions) && !ctx.settings.YshowPrintErrors.value =>
            val msg = ex match { case te: TypeError => te.toMessage case _ => ex.getMessage }
            s"[cannot display due to $msg, raw string = $x]"
      case _ => String.valueOf(x)
```

Reply:
I tried this out and it seems that we only use the `showAny` variant of this method. Not sure if we should keep `show`.

Reply:
Yeah, I added the defaulted Show at the last second, but I can't quite remember what the reason was.  It was niche though, perhaps something about in case a Seq behind a Shown arrived here.  Anyways, you're right about mixed concerns so I'm happy to lose the using clause.